backup/filebackend.py

```diff
@@ -95,13 +101,16 @@ def rewind(self):
         self.prev_version, self.offsets[1] = 0, 0
         return True
 
-    def stream_changes(self) -> Iterator[Change]:
-        self.read_metadata()
+    def stream_changes(self, stop_version=None, offset=None) -> Iterator[Change]:
+        if not stop_version and self.read_metadata():
+            stop_version = self.version
+
+        offset = 512 if offset is None else offset
         version = -1
         with open(self.url.path, 'rb') as f:
             # Skip the header
-            f.seek(512)
-            while version < self.version:
+            f.seek(offset)
+            while version < stop_version:
```

Comment:
`stop_version` is set to `self.version` in case it wasn't given as argument:
```
        if not stop_version and self.read_metadata():
            stop_version = self.version
```
But that can be made safer, like:
```
        if stop_version is None:
            self.read_metadata()
            stop_version = self.version
```
We also assume `read_metadata()` doesn't return `False`, seems reasonable.